src/Console/Shell.php

```diff
@@ -709,19 +709,35 @@ public function hr($newlines = 0, $width = 63)
      * Displays a formatted error message
      * and exits the application with status code 1
      *
+     * @param string|null $message Error message
+     * @param int $exitCode Exit code
+     * @return int Exit code
+     */
+    public function abort($message, $exitCode = 1) {
+        $this->_io->err('<error>' . $message . '</error>');
+        $this->_stop($exitCode);
```

Comment:
Our [backwards compatibility guide](http://book.cakephp.org/3.0/en/contributing/backwards-compatibility.html) doesn't mention anything about adding new exceptions. Given our current documented promises, converting exit() into an exception would be totally OK, and is something that I think is a net improvement.